demo/admin/src/generator/generateFormField.ts

```diff
@@ -0,0 +1,62 @@
+import { IntrospectionObjectType, IntrospectionQuery } from "graphql";
+
+import { FormConfig, FormFieldConfig, GeneratorReturn } from "./generator";
+import { camelCaseToHumanReadable } from "./utils/camelCaseToHumanReadable";
+import { Imports } from "./utils/generateImportsCode";
+
+export function generateFormField(
+    { gqlIntrospection }: { gqlIntrospection: IntrospectionQuery },
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    config: FormFieldConfig<any>,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    formConfig: FormConfig<any>,
+): GeneratorReturn & { imports: Imports } {
+    const gqlType = formConfig.gqlType;
+    const instanceGqlType = gqlType[0].toLowerCase() + gqlType.substring(1);
+
+    const name = String(config.name);
+    const label = config.label ?? camelCaseToHumanReadable(name);
+
+    const introspectionObject = gqlIntrospection.__schema.types.find((type) => type.kind === "OBJECT" && type.name === gqlType) as
+        | IntrospectionObjectType
+        | undefined;
+    if (!introspectionObject) throw new Error(`didn't find object ${gqlType} in gql introspection`);
+
+    const introspectionField = introspectionObject.fields.find((field) => field.name === name);
+    if (!introspectionField) throw new Error(`didn't find field ${name} in gql introspection type ${gqlType}`);
+
+    const requiredByIntrospection = introspectionField.type.kind == "NON_NULL";
+
+    const required = config.required ?? requiredByIntrospection; //if undefined default to requiredByIntrospection
+
+    //TODO verify introspectionField.type is compatbile with config.type
+
+    const imports: Imports = [];
+    let code = "";
+    if (config.type == "text") {
```

Comment:
i thought it would be easier to maintain end extend if field-types were separated into different files, given some structure defining possible code-sections. because i think they will become more complex. for example a select field will need to add code to the component-init region (where all the useXXX calls are located) and maybe adjust the if for the error- and loading-fallback component.

Reply:
currently i'm missing the introspection infos in my implementation, but I think this should be definitively added as this is required or at least handy to improve the possibilities for code-generation of any field. (stuff like adding a correctly typed scope var, or probably detecting the name of the primary-key field)